frontend/packages/data-portal/app/routes/browse-data.datasets.tsx

```diff
@@ -93,17 +90,15 @@ export default function BrowseDatasetsPage() {
     setParams: setPreviousBrowseDatasetParams,
   })
 
+  const [, setSearchParams] = useSearchParams()
   const handleRemoveFilter = () => {
-    setFromLocation(null)
-    setOrganism(null)
-    setDepositionId(null)
```

Comment:
this was causing a race condition because we were essentially calling `setSearchParams()` multiple times at once. the solution was to do it in one go via batch call to `setSearchParams()`. this is unfortunately a limitation with `useQueryParam()` and at the moment, we might need to refactor it or deprecate it entirely since it can lead to issues like this